From: Kevin Hilman <kevin@hilman.org>
Subject: [PATCH/RFC -rt] local_bh_enable() is safe for irqs_disabled()

In local_bh_enable() there is a WARN_ON(irqs_disabled()), but looking
at the rest of the code, it seems it expects to be used with
interrupts off, so is this warning really necessary?

I hit this warning in the ads7846 touchscreen driver timer function
where enable_irq() may be called with interrupts disabled.  Since
enable_irq now calls local_bh_disable/enable for IRQ resend, this
warning is triggered.

Patch against 2.6.23.9-rt12

Signed-off-by: Kevin Hilman <khilman@mvista.com>

---
 kernel/softirq.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-2.6.26/kernel/softirq.c
===================================================================
--- linux-2.6.26.orig/kernel/softirq.c
+++ linux-2.6.26/kernel/softirq.c
@@ -209,7 +209,6 @@ void local_bh_enable(void)
 
 	WARN_ON_ONCE(in_irq());
 #endif
-	WARN_ON_ONCE(irqs_disabled());
 
 #ifdef CONFIG_TRACE_IRQFLAGS
 	local_irq_save(flags);
